pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -193,14 +196,24 @@ func (cs *ConntrackConnectionStore) addNetworkPolicyMetadata(conn *flowexporter.
 // or adds a new connection with the resolved K8s metadata.
 func (cs *ConntrackConnectionStore) AddOrUpdateConn(conn *flowexporter.Connection) {
 	connKey := flowexporter.NewConnectionKey(conn)
-	cs.mutex.Lock()
-	defer cs.mutex.Unlock()
-	existingConn, exists := cs.connections[connKey]
+	cs.Mutex.Lock()
+	defer cs.Mutex.Unlock()
+
+	cs.expirePriorityQueue.Mutex.Lock()
+	defer cs.expirePriorityQueue.Mutex.Unlock()
```

Comment:
Move these closer to the priority queue operations in if-else blocks?

Reply:
Are you implying that using the lock twice separately in these two if-else blocks? If that's better I can do that.

Reply:
It is not critical. It just makes locking efficient by minimizing critical section.